controllers/config/clientconfig_controller.go

```diff
@@ -0,0 +1,215 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/clientcache"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/utils"
+)
+
+const (
+	ClientConfigHashAnnotation = k8ssandraapi.ResourceHashAnnotation
+	KubeSecretHashAnnotation   = "k8ssandra.io/secret-hash"
+)
+
+type ClientConfigReconciler struct {
+	Scheme       *runtime.Scheme
+	ClientCache  *clientcache.ClientCache
+	shutdownFunc context.CancelFunc
+
+	// filterMutex  sync.RWMutex
+	secretFilter map[types.NamespacedName]types.NamespacedName
+}
+
+/*
+	What about situations:
+		- Remove clientconfig when the cluster is still added?
+			* Cluster properties get updated?
+				* Do we accept or not?
+			* Could just be a cert-update etc, short-term, long-term ones
+		- New ClientConfig doesn't work?
+		- Add cluster points to a clientconfig that doesn't exist?
+		- What if the ClientConfig's kubeConfig is changed in the secret?
+		- What if certs expire? What if client connection fails - how do we detect when we need to do something to clientConfig?
+		- Do we load clientConfigs from all watch namespaces (or empty - cluster wide) or just from where operator is installed?
+			- Security /  user rights? If a namespace is allowed to add a K8ssandraCluster, but not new Kubernetes clusters to connect to?
+*/
+
+func (r *ClientConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	clientConfig := configapi.ClientConfig{}
```

Comment:
This isn't handling deletions correctly. If I delete a ClientConfig, the `Get` call returns an IsNotFound error. We need to check the error type and call `shutdownFunc` when it is IsNotFound. 

Since we aren't adding a finalizer, I don't think it is necessary to check the deletion timestamp.